core/primitives/src/challenge.rs

```diff
@@ -33,8 +33,15 @@ pub struct ChunkProofs {
     pub block_header: Vec<u8>,
     /// Merkle proof of inclusion of this chunk.
     pub merkle_proof: MerklePath,
-    /// Invalid chunk in encoded form.
-    pub chunk: EncodedShardChunk,
+    /// Invalid chunk in encoded form or in decoded form.
+    pub chunk: MaybeEncodedShardChunk,
+}
+
+/// Either `EncodedShardChunk` or `ShardChunk`. Used for `ChunkProofs`.
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Clone, Debug)]
+pub enum MaybeEncodedShardChunk {
```

Comment:
why do we need this?

Reply:
It's to avoid rencoding the chunk when it's already decoded.